### PR TITLE
feat: 344 persisted fetch viewer stars

### PR DESCRIPTION
**Pull request by bdougie**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

Adding a recommended repository section (Last 3 starred repos). Trying to keep the scope specific to adding star data. In a future PR, I will focus on making this more useful. 

Future features updates to address in a different pr
- remove repos that have already been added from the recommendation.
- load 15 of the last starred repos
- source star data from other users

#344 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

Screen Shot 2021-05-25 at 10 58 04 PM

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by filiptronicek** [comment]
@bdougie seems that this is a WIP, can we convert it into a draft?

**Comment by bdougie** [comment]
Please do not review this PR, it has a vicious error that will mess up your data :)

note to self: 
- Need to save repo full names and not URLs. 
- Need to create repo outside of the rendering loop.

File: src/components/RecommendedRepoList.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+              src={`https://avatars.githubusercontent.com/${goal.nameWithOwner.split("/")[0].replace(/\s+/g, "")}`}
+            />
+            <Flex className="details">
+              <p>{truncate(goal.nameWithOwner, 60)}</p>
+            </Flex>
+          </FlexCenter>
+        </FloatLeft>
+        <FloatRight>
+          <FlexCenter>
+            <a onClick={_handleGoalCreation(goal.nameWithOwner)} href="#"><img alt="pointing right icon" src={plus} /></a>
```
Comment (bdougie) [inline src/components/RecommendedRepoList.js:38]: Everytime the page is loaded it adds the rpeo. 
```suggestion
            <a onClick={() => _handleGoalCreation(goal.nameWithOwner)} href="#"><img alt="pointing right icon" src={plus} /></a>
```

File: src/components/DangerZone.js [hunk 2]
```diff
@@ -11,6 +11,7 @@ function DangerZone({goalId, repoName, note}) {
   const [removeModalOpen, setRemoveModalOpen] = useState(false);
 
   const _handleRepoRemoval = () => {
+    // TODO: resource not accessible by integration
```
Comment (bdougie) [inline src/components/DangerZone.js:14]: This is due to the rate limit being hit by testing this PR ironically.

**Comment by bdougie** [comment]
This corrupting data sadly :(

**Comment by bdougie** [comment]
Ok, it seems like it was a cache thing. Everything is working as expected.

**Pull request merged**